main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+type Command struct {
+	Name     string
+	Argument string
+}
+
+func readInput() (<-chan Command, []string) {
+	cmds := stdIn()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var filePath string
+	out := make(chan Command, len(cmds))
+
+	go func(cmds []string, filePath *string) {
+		for i, cmd := range cmds {
+			sepCmd := Split(cmd)
+			if i == 0 && len(sepCmd) == 1 {
+				log.Fatal("filepath not given")
+			}
+			command := Command{}
+			for k, name := range sepCmd {
+				if command.Name == "" {
+					command.Name = strings.ToLower(name)
+				} else if name != "" {
+					if *filePath == "" && k == len(sepCmd)-1 {
+						*filePath = name
+						wg.Done()
+						continue
+					} else {
+						command.Argument = name
+					}
+				}
+			}
```

Comment:
Put for loops in functions, nesting is too big in this place. Also I think you can simplify if statements

Reply:
Agreed